pandas/core/groupby/groupby.py

```diff
@@ -2633,6 +2633,93 @@ def _reindex_output(
 
         return output.reset_index(drop=True)
 
+    def sample(
+        self,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool = False,
+        weights=None,
+        random_state=None,
+    ):
+        """
+        Return a random sample of items from each group.
+
+        You can use `random_state` for reproducibility.
+
+        .. versionadded:: 1.1.0
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of items to return. Cannot be used with `frac`.
+            Default = 1 if `frac` is None.
+        frac : float, optional
+            Fraction of items to return. Cannot be used with `n`.
+        replace : bool, default False
+            Allow or disallow sampling of the same row more than once.
+        weights : list-like, optional
+            Default None results in equal probability weighting.
+            If passed a list-like then values must have the same length as
+            the underlying object and will be used as sampling probabilities
+            after normalization within each group.
+        random_state : int, array-like, BitGenerator, np.random.RandomState, optional
```

Comment:
It should be a Generator, not a BitGenerator. BitGenerator is the lower-level interface that only exposes the raw pseudo random numbers.  Also, are int or array-like seed values used with a RandomState or with a new BitGenerator?  Ideally one would limit the use of RandomState to end-user- provided RS and would use `np.random.default_rng([int, array_like])` if available.

Reply:
Also, might be clearer to call the just `state` since `random_state` gives the impression that it is linked to a `RandomState`.

Reply:
> It should be a Generator, not a BitGenerator.

You could be right, I was honestly lifting this docstring for the most part from the one for NDFrame.sample. If this is not correct we'd need to update it there as well as the random_state function in /core/common.py.

> Also, might be clearer to call the just state since random_state gives the impression that it is linked to a RandomState.

I feel we'd want to share the same naming convention used by NDFrame.sample and sklearn.

Reply:
Ahh, I misread it.  It does take a BitGenerator, which is sort of like a seed.  So it is correct. I understand the point of wanting compact.  I is unfortunate that `random_state` has become so ubiquitous since it makes moving to better practices or better rngs more difficult.